MathLib/Point3dWithID.h

```diff
@@ -63,6 +63,8 @@ class Point3dWithID: public Point3d {
 
     std::size_t getID() const { return _id; }
 
+    void setID(const std::size_t id) { _id = id; }
```

Comment:
The `id` can be set via constructor. No need for this public method that breaks encapsulation.